File: DNAseq/Models/Chromosome.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DNAseq.Models
{
    public class Chromosome
    {
        List<Gene> geneList;
        //chainName is the defining name for the list of genes that have common traits
        //such as collection of Height genes, collection of Color genes and so on
        string chainName;

        public Chromosome(string chainName)
        {
            geneList = new List<Gene>();
            this.chainName = chainName;
        }

        public List<Gene> getGenes()
        {
            return new List<Gene>(geneList);
        }

        public void addGene(Gene gene)
        {
            this.geneList.Add(gene);
        }

        public string getChainName()
        {
            return chainName;
        }
    }

    

    
}

File: scripts/Chromosome.js
import Gene from './Gene.js';

export default class Chromosome {
    constructor(chainName) {
        this.geneList = [];
        this.chainName = chainName;
    }

    getGenes() {
        return this.geneList;
    }

    addGene(gene) {
        this.geneList.push(gene);
    }

    getChainName() {
        return this.chainName;
    }
    
    
}
